app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CompaniesController < ApplicationController
+
+  before_action :authenticate_user!
+  before_action :find_company_by_id, only: [:show, :edit, :update, :destroy]
```

Comment:
@AmiFaldu-BTC : we can use it like ... except: [:new] instead of using only & writing 4 actions